_posts/2018-10-01-angular-state-management-comparison.md

```diff
@@ -0,0 +1,428 @@
+The next question is how this could work performant and reliable.
+
+The current trend is to use a [Redux](https://redux.js.org/)-based storage solution, which consists of a Store, Selectors to get data from the store in the form of Observables and Actions to modify the store.
+This allows for a single source of truth, a read-only state and the flow of date going in one direction.
+There are a couple of different solutions for Angular.
+NGRX is by far the most popular, leaving the new kids in town, NGXS and Akita, far behind in popularity.
+
+It is, however, not always needed to have a storage framework solution.
+Very small applications are easy to create with plain RxJS, if you are quite skilled.
+In this post, I've stacked each of these solutions against each other to see what can be learned.
+
+# Table of content
+
+1. [The fighting ring](#the-fighting-ring)
+2. [The competitors](#the-competitors)
+3. [Fight](#fight)
+    1. [Available Tooling](#1-available-tooling)
+    2. [Features](#2-features)
+    3. [Boilerplate code](#3-boilerplate-code)
+    4. [Community](#4-community)
+    5. [Dependencies and size](#5-dependencies-and-size)
+4. [Final score](#final-score)
+
+# The fighting ring
+
+To compare the four competitors, I've set up a simple To Do-application ([GitHub](https://github.com/orjandesmet/angular-state-management-comparison)) with **Angular CLI**.
+
+The master branch holds a base of the application, which cannot run on its own.
+It needs a state management solution from one of the other branches.
+To make the comparison easier, the base application is written in such a way that each solution only adds files to a *statemanagement* folder and loads a service and zero or more modules into the **AppModule**.
+No other files (except *package.json*, *package-lock.json* and *logo.png*) are to be changed.
+From an end-user perspective, the application would appear and behave the exact same, no matter which state management solution is used.
+The logo is added to be able to differntiate which solution is running.
+
+A To Do-application is perfect to demonstrate CRUD.
+A **FakeBackendService** is provided to simulate a RESTful api backend.
+The idea is to only load the list only once in the application's lifetime and then update the state, without needing to fetch everything from the backend again.
+As such, the **FakeBackendService** logs its calls to the console for monitoring.
+
+# The competitors
+
+### NGRX
+
+(v6.1.x) [Docs](https://github.com/ngrx/platform/blob/master/README.md)
+
+> RxJS powered state management for Angular applications, inspired by Redux
+>
+> @ngrx/store is a controlled state container designed to help write performant, consistent applications on top of Angular.
+
+### NGXS
+
+(v3.2.x) [Docs](https://ngxs.gitbook.io/ngxs/)
+
+> NGXS is modeled after the CQRS pattern popularly implemented in libraries like Redux and NGRX but reduces boilerplate by using modern TypeScript features such as classes and decorators.
```

Comment:
Modelled

Reply:
This is actually a quote from their website and correct in American English